app/diagnostics.go

```diff
@@ -716,6 +714,11 @@ func (a *App) trackPlugins() {
 			if plugin.Manifest.SettingsSchema != nil {
 				settingsCount += 1
 			}
+
+			pluginTrackData[plugin.Manifest.Id] = map[string]interface{}{
```

Comment:
When I read this without the context below, I thought we were starting to track all plugin IDs, and not just the whitelisted ones. 

I see now that we're just preprocessing this data for use below... but I wonder if we shouldn't just make `pluginData` accept the `pluginStates` and available plugins directly and do it's own processing. A few more iterations through the data, perhaps, but the overhead will be negligible, and may bring clarity here.

It might also eliminate the point @hanzei raises re: a new use of `interface{}`.

Reply:
Sure we can do that. My concern was that `pluginsEnvironment.Available()` does a file store folder scan to get the available plugins and wanted to avoid the repeated I/O operation for each plugin each time the diagnostics data is sent that I am not sure how often it is done.

Reply:
Agree that we should avoid multiple calls to `Available()` -- definitely just cache that once and reuse.